services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_scheduler.py

```diff
@@ -553,32 +569,20 @@ async def _run_trigger_observation_queue_task(self) -> None:
         logger.info("Scheduler 'trigger observation queue task' was shut down")
 
     async def _run_scheduler_task(self) -> None:
-        settings: DynamicServicesSchedulerSettings = (
-            self.app.state.settings.DYNAMIC_SERVICES.DYNAMIC_SCHEDULER
-        )
-        logger.debug(
-            "dynamic-sidecars observation interval %s",
-            settings.DIRECTOR_V2_DYNAMIC_SCHEDULER_INTERVAL_SECONDS,
-        )
-
-        while self._keep_running:
-            logger.debug("Observing dynamic-sidecars %s", list(self._to_observe.keys()))
+        logger.debug("Observing dynamic-sidecars %s", list(self._to_observe.keys()))
 
-            try:
-                # prevent access to self._to_observe
-                async with self._lock:
-                    for service_name in self._to_observe:
-                        self._enqueue_observation_from_service_name(service_name)
-            except asyncio.CancelledError:  # pragma: no cover
-                logger.info("Stopped dynamic scheduler")
-                raise
-            except Exception:  # pylint: disable=broad-except
-                logger.exception(
-                    "Unexpected error while scheduling sidecars observation"
-                )
-
-            await sleep(settings.DIRECTOR_V2_DYNAMIC_SCHEDULER_INTERVAL_SECONDS)
-            self._observation_counter += 1
+        try:
+            # prevent access to self._to_observe
+            async with self._lock:
+                for service_name in self._to_observe:
+                    self._enqueue_observation_from_service_name(service_name)
+        except asyncio.CancelledError:  # pragma: no cover
+            logger.info("Stopped dynamic scheduler")
```

Comment:
Why did you log as info here ? And if so, is there a way to get more infomation on why+who was the task cancelled?

Reply:
Now that I look at it better. It makes no sense. This period task is actually cancelled and it will lead to tracebacks. Dropping this error which is not used